docs/tech_specs/monitoring_data_consumption.md

```diff
@@ -0,0 +1,61 @@
+  2. Set up an endpoint for ARINC to send requests to CTD informing whether a given sign is up or down at some frequent interval (something like every 30 seconds – 1 Min)
+
+The order in which these goals are accomplished is arbitrary, or they could be worked on in parallel.
+
+## Proposed solution
+### Part 1: Message Logs 
+
+To give more context on the reasoning behind the solution proposed, we should note that Realtime Signs runs on `opstech3` which has historically been prone to run out of memory and experience other performance issues. It's also worth noting that opstech3 is intended to be a DB server and not meant to host applications like Realtime Signs. Therefore, we should try to keep the footprint of whatever new logic/features we add to Realtime Signs as minimal. Also, it is to our advantage to host processing-heavy logic in AWS since that allows us to scale resources if we need to. We don't have that ability on opstech3.
+
+To give a rough idea of the required resources for processing the message logs, the CSV files may be up to ~200MB which is based on the number of messages we send to ARINC over the course of a day. While this ultimately isn’t a ton of data, we should note that this is only counting visual messages. We also send audio messages, but they are not able to be logged yet. If/when those are added, the number of rows in the CSV would at least double.
+
+As a list of steps, the solution would involve the following in sequence: 
+
+  1. ARINC builds out an application that filters out unneeded logs and compiles the data that we outlined in our specification. With CTD having improved the usability of the `uid` field that we send in our messages to the ARINC headend server, this is achievable now.
+
+  2. ARINC sets up an endpoint for CTD to hit once per day around 2AM to receive a file containing the compiled logs. It would probably be good if the file is a zip to keep the size of data being transferred between servers small.
+
+  3. CTD sets up a recurring job which calls ARINC’s endpoint nightly to receive the file and dump it into an S3 bucket. Having the logs stored in S3 also makes it convenient if we ever need to dig up past log files for some reason.
+
+  4. CTD sets up an application that is responsible for pulling down the zip file from S3, unzipping it, and then streaming it into Splunk. There may be different options to explore on how to achieve this step. Splunk does not provide an easy way of uploading a CSV file through their API. They do provide a mechanism called a Universal Forwarder meant for forwarding large amounts of data into Splunk, though past attempts at setting these up have not been successful at the CTD. A "brute force" option could be to stream the CSV and `Logger.log()` the rows into Splunk, but the amount of requests would be close to 600,000 for one CSV for visual messages only.
```

Comment:
I'd recommend something like Lambda or Batch for this, rather than a new ECS application, as doesn't need to be running all of the time. Lambda would be a Node or Python script, and Batch would be anything that can go into a Docker container.

You don't need to make a single request for each log line: they can be batched. See example 3 in the [Splunk documentation](https://docs.splunk.com/Documentation/Splunk/latest/Data/FormateventsforHTTPEventCollector) for the HTTP Event Collector. 

I'm not sure how many you could combine in a single request, but possibly it's all of them?